tests/configurable_cluster.py

```diff
@@ -194,29 +208,46 @@ def copy_to_host(self, source_path, host, dest_path=None):
             dest_path = os.path.join(self.mount_dir,
                                      os.path.basename(source_path))
         self.exec_cmd_on_host(host, 'mkdir -p ' + os.path.dirname(dest_path))
+        self.exec_cmd_on_host(host, 'chown {0}:{0} {1}'.format(self.user, os.path.dirname(dest_path)))
 
         ssh = paramiko.SSHClient()
         ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy())
-        ssh.connect(host, username=self.user, password=self.password,
+        ssh.connect(host, username=self.user, key_filename=self.key_path,
                     timeout=180)
+
+        # Upload to dummy location because paramiko doesn't allow SFTP using
+        # sudo when logged in as a non root user
+        dummy_path = '/tmp/{0}/{1}'.format(str(uuid.uuid1()), os.path.basename(dest_path))
+        self.exec_cmd_on_host(host, 'mkdir -p {0}'.format(os.path.dirname(dummy_path)))
+        self.exec_cmd_on_host(host, 'chown {0}:{0} {1}'.format(self.user, os.path.dirname(dummy_path)))
         sftp = ssh.open_sftp()
-        sftp.put(source_path, dest_path)
+        sftp.put(source_path, dummy_path)
         sftp.close()
+
+        # Move to final location using sudo
+        self.exec_cmd_on_host(host, 'mv {0} {1}'.format(dummy_path, dest_path))
+
+        # Remove dummy path directory
+        self.exec_cmd_on_host(host, 'rm -rf {0}'.format(os.path.dirname(dummy_path)))
+
         ssh.close()
 
+    # Since ConfigurableCluster is configured using external IPs, those act as
```

Comment:
The cluster.master field is supposed to return an IP or hostname that is externally accessible from anywhere. Of course because we directly fetch that variable, there's no way to enforce it between implementations (that should change with a refactor I have in mind).

cluster.master should be resolvable from the machine running the tests and in the case of configurable cluster that happens to be my machine/jenkins which will use the external interface.